examples/02_miscellaneous/plot_1_pearson_correlation.py

```diff
@@ -0,0 +1,207 @@
+    target_features, axis=1
+).reshape(-1, 1)
+source_geometry_normalized = source_geometry / np.max(source_geometry)
+target_geometry_normalized = target_geometry / np.max(target_geometry)
+
+
+# %%
+# We first define a function to compute the Pearson correlation
+# between two tensors. Such function is not available in PyTorch,
+# but it is easy to implement.
+def pearson_corr(x, y):
+    """Compute the Pearson correlation between two tensors."""
+    if not torch.is_tensor(x) or not torch.is_tensor(y):
+        x = torch.tensor(x)
+        y = torch.tensor(y)
+
+    vx = x - torch.mean(x)
+    vy = y - torch.mean(y)
+
+    corr = (
+        torch.sum(vx * vy)
+        / (torch.sqrt(torch.sum(vx**2)) * torch.sqrt(torch.sum(vy**2)))
+    ).numpy()
+
+    return corr
+
+
+# %%
+# We then define a callback function that computes the
+# Pearson correlation between the transformed features and
+# the target features at each BCD iteration.
+
+# Initialize the list of Pearson correlations with the original
+# untansformed features
+corr_bcd_steps = [
+    pearson_corr(source_features_normalized, target_features_normalized)
+]
+
+
+def correlation_callback(
+    locals,
+    source_features=None,
+    target_features=None,
+    device=torch.device("cpu"),
+):
+    console = Console()
+
+    # Get the current transport plan and tensorize the features
+    pi = locals["pi"]
+    source_features_tensor = _make_tensor(source_features, device)
+    target_features_tensor = _make_tensor(target_features, device)
+
+    # Compute the transformed features
+    transformed_features = (
+        (pi @ source_features_tensor.T / pi.sum(dim=1).reshape(-1, 1))
+        .T.detach()
+        .cpu()
+    )
+
+    # Compute the Pearson correlation and append it to the list
+    corr = pearson_corr(transformed_features, target_features_tensor)
+    corr_bcd_steps.append(corr)
+
+    console.log("Pearson correlation: ", corr)
+
+
+# %%
+# We now fit the mapping using the sinkhorn solver and 10 BCD iterations.
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+mapping = FUGW(alpha=0.5, rho=1, eps=1e-4)
+_ = mapping.fit(
+    source_features=source_features_normalized,
+    target_features=target_features_normalized,
+    source_geometry=source_geometry_normalized,
+    target_geometry=target_geometry_normalized,
+    solver="sinkhorn",
+    solver_params={
+        "nits_bcd": 10,
+    },
+    callback_bcd=partial(
+        correlation_callback,
+        source_features=source_features_normalized,
+        target_features=target_features_normalized,
+        device=device,
+    ),
+    verbose=True,
+)
+
+# %%
+# We finally plot the evolution of the Pearson correlation and
+# the training loss at each BCD iteration. We can see that the
+# Pearson correlation dips at the first iteration, but quickly
+# increases and stabilizes at the second iteration. This
+# is due to the initialization of the transport plan, which
+# is filled with ones.
```

Comment:
This comment will probably no longer be true after you make updates related to my previous point